### PR TITLE
UPD Link Fix

### PR DESCRIPTION
**Pull request by dogmaphobic**
Fixes to the UDP link.

This is in response to issue #1371. I got my MavLink *slave* done (the most convoluted pile of wires with a Raspberry PI in the middle of it all) and was able to finally test UDP links. In the process, I found not only the issue mentioned above but also a couple of thread racing conditions. UPD links are functional but as I only have one MavLink device, I have not tested it with multiple connections.

### CODE REVIEW

File: src/comm/UDPLink.cc [hunk 1]
```diff
@@ -317,32 +312,43 @@ void UDPConfiguration::addHost(const QString& host)
             for (int i = 0; i < hostAddresses.size(); i++)
             {
                 // Exclude loopback IPv4 and all IPv6 addresses
-                if (!hostAddresses.at(i).toString().contains(":"))
+                if (!hostAddresses.at(i).toString().contains(":") && !!hostAddresses.at(i).toString().startsWith("127"))
```
Comment (LorenzMeier) [inline src/comm/UDPLink.cc]: I'm not too sure about this one. We have quite often running the autopilot code and the GCS on the same machine. Could you think of any reason to disallow localhost here?

**Comment by LorenzMeier** [comment]
Looks good, except for the line excluding localhost.

**Comment by dogmaphobic** [comment]
> Could you think of any reason to disallow localhost here?

Not really. The only reason I did that was because of the original comment in the code ( which I ended up removing in the process).

**Comment by dogmaphobic** [comment]
Loopback re-enabled.

**Pull request merged**